### PR TITLE
Offers: Initial Docs

### PR DESCRIPTION
**Pull request by kpollich**

### CODE REVIEW

File: business-locations.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Fields
+
+| Name | Type | Descriptions |
+|------|------|--------------|
+| latitude | decimal | The location's latitude in degrees |
+| longitude | decimal | The location's longitude in degrees |
+| slug | string | The URL partial for the location |
+| address | string | Street address for the location |
+| address2 | string | Additional address fields for the location |
```
Comment (scottgonzalez) [inline business-locations.md]: s/fields/field/

File: businesses.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Businesses
+
+## Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| name | string | Name of the business |
+| slug | string | Formerly "SEOUrl" - the url partial for the business |
+| is_vanity | boolean | Whether this business is only a customer for TLS custom apps |
+| esp_business_id | integer | Billing account ID number for the business |
```
Comment (scottgonzalez) [inline businesses.md]: Let's expose this as just `esp`.

File: deals.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Deals
+
+## Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| legacy_deal_id | integer | The deal's ID in old systems |
```
Comment (scottgonzalez) [inline deals.md]: Do we need to expose this?

File: deals.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+| certificate_valid_for_days | integer | Number of days remaining for which the deal is valid |
```
Comment (scottgonzalez) [inline deals.md]: This isn't days remaining.

File: deals.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+| original_price | decimal | Pre-discount price |
+| offer_price | decimal | Post-discount price |
+| buck_rebate | decimal | Amount of gift bucks returned for purchasing the deal |
```
Comment (scottgonzalez) [inline deals.md]: s/returned/earned/

File: deals.md (hunk 3 above)
Comment (kpollich) [inline deals.md]: Doubtful. I'd rather not expose it.

File: deals.md (hunk 4 above)
Comment (kpollich) [inline deals.md]: Oh I thought this was a maintained column. @Mamsler - is this just total number of days for which the cert is valid?

File: deals.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+| max_quantity_per_customer | integer | Maximum deals purchasable per customer |
+| max_gift_quantity_per_customer | integer | Maximum deals available for gifting per customer |
+| split | integer | Factor by which to split the deal into cheaper deals |
+| slug | string | Partial URL for the deal |
+| title | string | Primary title for the deal |
+| use_now | boolean | Whether the deal can be used immediately |
+| no_affiliate | boolean | ? |
+| is_national_deal | boolean | Whether the deal runs at a national (not local) level |
+| requires_shipping_information | boolean | Whether the deal requires shipping information to purchase |
+| is_discoverable | boolean | ? |
```
Comment (scottgonzalez) [inline deals.md]: Whether the deal is displayed in categories and search results.

Some deals go live but require a direct link. I think this is done when merchants want to provide exclusive access to some of their deals.

File: deals.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+| show_remaining_quantity | boolean | Whether the deal displays the number of deals remaining that are available for purchase |
+| hide_from_site | boolean | Whether the deal is hidden from the site |
```
Comment (scottgonzalez) [inline deals.md]: Is this still supposed to exist?

Comment (kpollich) [inline deals.md]: I suppose deals that are hidden from the site shouldn't be in the `offer` database at all, huh?

File: deals.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+| start_date | date | The date at which the deal is first available for purchase |
+| affiliate_start_date | date | ? |
+| end_date | date | The date at which the deal is no longer available for purchase |
+| close_date | date | The date at which the deal no longer appears on the site |
+| certificate_expiration_date | date | The date at which the certificate provided when the deal is purchased expires |
```
Comment (scottgonzalez) [inline deals.md]: Perhaps less confusing wording like "The date at which the deal may no longer be redeemed"

File: deals.md (hunk 4 above)
Comment (scottgonzalez) [inline deals.md]: My understanding is that this is mutually exclusive with `certificate_expiration_date`. Either the cert is good until a specific date or it's good for X days after purchase.

File: deals.md (hunk 7 above)
Comment (scottgonzalez) [inline deals.md]: I think `is_discoverable` replaced this.

Comment (Mamsler) [inline deals.md]: i had this discussion with Justin. he said include it.

File: deals.md (hunk 4 above)
Comment (Mamsler) [inline deals.md]: ture

File: deals.md (hunk 7 above)
Comment (scottgonzalez) [inline deals.md]: What does this control?

Comment (Mamsler) [inline deals.md]: i thought is_discoverable had to do with affiliated being able to view it

Comment (scottgonzalez) [inline deals.md]: Honestly, I don't even know what affiliates are in our app.

Comment (Mamsler) [inline deals.md]: hide_from_site is like a temporary force closed. there might be a problem with the deal so they hide it untill they can either fix it or know that they can force close it

Comment (Mamsler) [inline deals.md]: i think the only affiliate we have right now is commission junction. basically some other sites show are content

Comment (scottgonzalez) [inline deals.md]: Why is that not controlled by status?

File: deals.md (hunk 8 above)
Comment (Mamsler) [inline deals.md]: i don't really think certificate_expiration_date is confusing

Comment (scottgonzalez) [inline deals.md]: I was commenting on the description.

**Comment by kpollich** [comment]
I haven't listed example requests for deals, coupons, or business locations. I don't know that we need to implement individual endpoints for those resources in our .NET services. The only offer endpoints -  @pperera9 @Ch3rub correct me if I'm wrong - I think is the businesses endpoint (with supported filters for latitude and longitude)

**Comment by kpollich** [comment]
This PR is no longer WIP and is ready for final review + merge. 

Note: I haven't done anything with `merchandise_deals` yet. @Mamsler is that schema still pretty volatile or should I start writing up docs?

**Pull request closed without merging**